### ISSUE TITLE
Exception while fetching data if query depth > 3 

### ISSUE THREAD

**Issue by DartVerder** [Haulmont/jmix-graphql#160]
**Environment:**
platform version: 1.0.0-SNAPSHOT

**Steps:**
1. Create NewEntity1 with attribute: name
2. Create NewEntity2 with attribute: name, composition to NewEntity1
3. Create inverse composition attribute in the NewEntity2 for the NewEntity1
4. Create instances of NewEntity2 and associated NewEntity1
5. 
```
{
  NewEntityList {
    id
    newEnt1 {
      newEntity {
        name
        newEnt1 {
          name
        }
      }
    }
  }
}

```

**AR:**
after adding last level `  newEnt1 {  name }` an error displayed in responce
```
{
  "errors": [
    {
      "message": "Exception while fetching data (/NewEntityList) : Cannot get unfetched attribute [the1] from detached object com.company.untitled3.entity.NewEntity-95a438f7-8d48-8ee0-169a-5f1a40fd9cd3 [detached].",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "NewEntityList"
      ],
      "extensions": {
        "classification": "DataFetchingException"
      }
    }
  ],
  "data": {
    "NewEntityList": null
  }
}
```

Note: it is important for query depth testing
[exception1.txt](https://github.com/Haulmont/jmix-graphql/files/6708151/exception1.txt)

Development notes:
Looks like issue reproduced when in nested level there is the same entity object with larger set of fields than in upper level.

Works fine:
```
{
  scr_CarList {
    garage {
      name
      cars {
        garage {
          name
        }
      }
    }
  }
}
```
But failed:
```
{
  scr_CarList {
    garage {
      cars {
        garage {
	  name
        }
      }
    }
  }
}
```

```
```

**Comment by vadimbasko** [comment]
Reproduced in `scr-jmix`
```
{
  scr_CarList {
    garage {
      cars {
        garage {
          cars {
            regNumber
          }
        }
      }
    }
  }
}
```

**Comment by vadimbasko** [comment]
Blocked by [jmix-data/issues/83](https://github.com/Haulmont/jmix-data/issues/83)

**Comment by dtaimanov** [comment]
Fixed in Haulmont/jmix-data#83
